.github/workflows/build-projects.yaml

```diff
@@ -0,0 +1,57 @@
+name: Build projects
+on: [push]
```

Comment:
I think it should be based on pull request

pull_request:
branches:
- main
push:
branches:
- main